src/snowflake/snowpark/modin/plugin/_internal/index.py

```diff
@@ -1519,10 +1524,19 @@ def to_series(self) -> None:
         Index.to_frame : Convert an Index to a DataFrame.
         Series.to_frame : Convert Series to DataFrame.
         """
-        # TODO: SNOW-1458117 implement to_series
+        from snowflake.snowpark.modin.pandas import Series
 
-    @index_not_implemented()
-    def to_frame(self) -> None:
+        pandas_index = self.to_pandas()
+        if index is None:
+            index = pandas_index
+
+        if name is None:
+            name = self.name
+
+        return Series(data=pandas_index, index=index, name=name)
+
+    # TODO: SNOW-1481037 : Fix typehints
```

Comment:
Yes the goal is to fix the typehints in a new PR. I would like to fix them in the current PRs but it is more challenging than it looks because of circular imports and the linter removing any object names in quotes